src/emit/expression.rs

```diff
@@ -2147,6 +2149,16 @@ pub(super) fn expression<'a, T: TargetRuntime<'a> + ?Sized>(
             advanced.into()
         }
 
+        Expression::PointerPosition { pointer } => {
+            let ptr = expression(target, bin, pointer, vartab, function, ns);
+            let data = bin.vector_bytes(ptr);
+            let res = bin
+                .builder
+                .build_ptr_to_int(data, bin.context.i64_type(), "sesa");
+
+            res.unwrap().into()
+        }
```

Comment:
Please help me understand `Expression::PointerPosition` . It takes a vector (e.g. `Type::DynamicString`), gets the pointer and converts the pointer to `i64` (even though pointers are 32 bit in wasm).

How about renaming it to `Expression::VectorData` ?

Reply:
The `i64` rationale is that Soroban host always takes `i64` values. And that expression is only used in Soroban at the moment